libvast/vast/system/reader_command_base.hpp

```diff
@@ -44,11 +44,12 @@ class reader_command_base : public node_command {
   using node_command::node_command;
 
 protected:
-  int run_impl(caf::actor_system& sys, option_map& options,
-               caf::message args) override;
+  int run_impl(caf::actor_system& sys, option_map& options, const_iterator,
+               const_iterator) override;
```

Comment:
What was the design decision behind an iterator pair as opposed to, say, `vector<string>`?

Reply:
A `const_iterator` pair allows us to remove already parsed arguments by simply increment the `begin` iterator. 
If we used a `vector<string>`, we would have to copy it a new vector for each sub-command.

Reply:
Okay, that makes sense.